content/tutorials/build-apps/build-a-desktop-wallet-in-javascript.md

```diff
@@ -0,0 +1,1503 @@
+
+This step shows how to add some checks on destination addresses to warn the user before sending XRP.
+
+One type of check we could make is to verify the domain name associated with an XRP Ledger address; this is called [account domain verification](xrp-ledger-toml.html#account-verification). When an account's domain is verified, we can could show it like this:
+
+![Screenshot: Step 8, use domain verification](img/javascript-wallet-8.png)
+
+1. In the `library` folder, add a new file `8_helpers.js`. Then add the following contents to that file:
+
+{{ include_code("_code-samples/build-a-wallet/desktop-js/library/8_helpers.js", language="js") }}
+
+The code in `8_helpers.js` looks up the account on the ledger by sending an [`account_info`](account_info.html) request.
+
+If the account does exist, the code checks for the [`lsfDisallowXRP` flag](accountroot.html#accountroot-flags).
+
+2. Import the new helper function in`index.js`:
+
+```javascript
+const { initialize, subscribe, saveSaltedSeed, loadSaltedSeed } = require('./library/5_helpers')
+const { sendXrp } = require('./library/7_helpers')
+// Step 8 code additions - start
+const { verify } = require('./library/8_helpers')
+// Step 8 code additions - end
+```
+
+3. After the callback function `ipcMain.on('send-xrp-action', callback)` add the following event handler:
+
+```javascript
+ipcMain.on('send-xrp-action', (event, paymentData) => {
+  sendXrp(paymentData, client, wallet).then((result) => {
+    appWindow.webContents.send('send-xrp-transaction-finish', result)
+  })
+})
+
+// Step 8 code additions - start
+ipcMain.on('destination-account-change', (event, destinationAccount) => {
+  verify(destinationAccount, client).then((result) => {
+    appWindow.webContents.send('update-domain-verification-data', result)
+  })
+})
+// Step 8 code additions - end
+```
+
+3. Modify `view/preload.js` and add the following two functions to `'electronAPI'`:
+
+```javascript
+onDestinationAccountChange: (callback) => {
+  ipcRenderer.send('destination-account-change', callback)
+},
+onUpdateDomainVerificationData: (callback) => {
+  ipcRenderer.on('update-domain-verification-data', callback)
+}
+```
+
+Finally, the code decodes the account's `Domain` field, if present, and performs domain verification using the method imported above.
+
+4. Update the view logic - in `view/template.html` add the following lines just before the `<input>` element with `id="input-destination-address`:
+
+```html
+<div class="input-group mb-3">
+    <!-- Step 8 code additions - start -->
+    <div class="accountVerificationIndicator">
+        <span>Verification status:</span>
+    </div>
+    <!-- Step 8 code additions - end -->
+    <input type="text" class="form-control" placeholder="rP4zcp52pa7ZjhjtU9LrnFcitBUadNW8Xz" id="input-destination-address">
```

Comment:
Nit: the placeholder address in this sample code is different than in the version from the previous example (in step 7).

```suggestion
    <input type="text" class="form-control" placeholder="r9jEyy3nrB8D7uRc5w2k3tizKQ1q8cpeHU" id="input-destination-address">
```